File: week_8_and_9/3_Ruby/fizz_buzz/my_solution.rb
# U3.W8-9: 


# I worked on this challenge with Amol Borcar and myself.

# 2. Pseudocode



# 3. Initial Solution


def super_fizzbuzz(array)
  fizzy=[]
	array.each do |x| 
	  if x%3==0 && x%5==0
		  fizzy << "FizzBuzz"
	  elsif x%5==0
		  fizzy << "Buzz"
	  elsif x%3==0
		  fizzy << "Fizz"
	  else
		  fizzy << x
	  end
	end
return fizzy
end



# 4. Refactored Solution






# 1. DRIVER TESTS/ASSERT STATEMENTS GO BELOW THIS LINE






# 5. Reflection 

File: week_8_and_9/2_JavaScript/translate_to_js/my_solution.js

// def is_fibonacci?(num)
// 	arr=[0,1]
//   	while num>arr[-1]
//   	arr << arr[-2]+arr[-1]
// end
//   	if num == arr[-1]
// 		return true
// end
// end

// U3.W8-9: Challenge you're converting


// I worked on this challenge 
//1. Andrew Koines Jr.
//2. Kendall Carey

// 2. Pseudocode
/*
Define a method called is_fibonacci that takes a number as an argument
 determines if the number given is in the fibonacci sequence and returns true 
or false depending.

Create a variable arr that is an array with two e.lements, 0 and 1

While the number argument is greater than the last element in the array,
Push the sum of the last two elements in the array onto the array

Return true if number is equal to the last element in the array.


*/
// 3. Initial Solution
var arr=[0,1];

is_fibonacci=function(num) {		
  	while (num>arr[arr.length-1]) {
  		arr.push(arr[arr.length-2]+arr[arr.length-1]);
  	}
  if (num==arr[arr.length-1]) {
    return true;
  }
};


// 4. Refactored Solution





// 1. DRIVER TESTS/ASSERT STATEMENTS GO BELOW THIS LINE

console.log(is_fibonacci(13)==true);
console.log(is_fibonacci(14)==false);




// 5. Reflection 
// This drove Andrew and I crazy when we thought it was going to be simple 
// since we already had the solution in Ruby. But it was crazy difficult!
// We even posted it on the junebug community group.